docs/src/main/asciidoc/config-reference.adoc

```diff
@@ -310,6 +310,52 @@ public class SomeBean {
 <1> At this injection point `greetingConfiguration` will use the `greeting` prefix since that is what has been defined on `@ConfigProperties`.
 <2> At this injection point `otherConfiguration` will use the `other` prefix from `@ConfigPrefix` instead of the `greeting` prefix. Notice that in this case `@Inject` is not required.
 
+=== Using a list of objects
+
+In certain cases it might be necessary to support complex configuration structures that utilize a list of objects as shown in the following example:
+
+`ComplexConfiguration.java`
+[source,java]
+----
+@ConfigProperties(prefix = "complex")
+public class ComplexConfiguration {
+    public String name;
+    public String user;
+    public String password;
+    public List<Nested> inputs;
+    public List<Nested> outputs;
+
+    public static class Nested {
+        public String user;
+        public String password;
+    }
+}
+----
+
+Support for such use cases is only possible when the `quarkus-config-yaml` extension is present. A corresponding example configuration could be:
```

Comment:
I think this is slightly misleading. I suppose it's only working if you are using Yaml configuration? The way it's worded, it looks like just having the extension around is enough.

Reply:
How about now?

Reply:
Mmmmh I don't see any difference?

Was thinking about something like:

> Support for such use cases is only available when using YAML configuration with the `quarkus-config-yaml` extension.

Reply:
I don't see the difference TBH, but I'll just copy what you mention and be done with ti

Reply:
Updated